packages/session/src/sessions.ts

```diff
@@ -62,6 +65,29 @@ export class Sessions implements ISessions {
       path: path ?? session.path,
       name: name ?? session.name,
     };
+
+    if (kernel) {
+      // Kernel id takes precedence over name.
+      if (kernel?.id) {
```

Comment:
Looks like the `?` here might not be needed if there is already a guard above with `if (kernel)`?